packages/ember-routing/lib/system/router.js

```diff
@@ -200,7 +200,9 @@ var EmberRouter = EmberObject.extend(Evented, {
     this._setupRouter(router, location);
 
     location.onUpdateURL((url) => {
-      this.handleURL(url);
+      if (this._shouldHandleURL(url)) {
```

Comment:
To me this feels like a location responsibility.  Can you move this into the locations and only call the `onUpdateURL` callback when we are handling the rootURL?